src/tree.c

```diff
@@ -293,22 +293,43 @@ static void ofi_delete_rebalance(struct ofi_rbmap *map, struct ofi_rbnode *node)
 	node->color = BLACK;
 }
 
+static void ofi_rbmap_exchange(struct ofi_rbmap *map,
+		struct ofi_rbnode *cur, struct ofi_rbnode *successor)
+{
+	if (successor == cur)
+		return;
+
+	*successor = *cur;
+
+	if (!cur->parent)
+		map->root = successor;
+	else if (cur == cur->parent->left)
+		cur->parent->left = successor;
+	else
+		cur->parent->right = successor;
+
+	if (cur->left != &map->sentinel)
+		cur->left->parent = successor;
+	if (cur->right != &map->sentinel)
+		cur->right->parent = successor;
+}
+
 void ofi_rbmap_delete(struct ofi_rbmap *map, struct ofi_rbnode *node)
 {
 	struct ofi_rbnode *x, *y;
```

Comment:
What do x and y stand for? Can they be renamed to meaningful ones?

Reply:
This is from the original rbtree code.  I did attempt to map those to more meaningful names, like successor, parent, predecessor, etc, but couldn't find anything that worked consistently.  For example, if you look at the code, x might be the right child, or the left child, or the right child of the leftmost descendant of the right child. I gave up at this point.  Maybe other names (e.g. successor, predecessor) could work if we define more variables, but I think x and y take on different roles based on the state of the tree.

Reply:
Got it. I was hoping renaming them would make it easy to follow the code. :)